From cc29efcc00afe3c3097c567d7003d5b072f92e60 Mon Sep 17 00:00:00 2001
From: Thomas Leonard <talex5@gmail.com>
Date: Sat, 17 May 2025 15:27:15 +0100
Subject: [PATCH] Fix incorrect type on write pipes

Crosvm was tagging pipes as CROSS_DOMAIN_ID_TYPE_VIRTGPU_BLOB instead of
CROSS_DOMAIN_ID_TYPE_WRITE_PIPE. This causes the guest proxy to attempt
to get a blob, which fails with e.g.

    wl-proxy [WARNING]: Error handling client: Unix.Unix_error(Unix.ENOSPC, "DRM_IOCTL_VIRTGPU_RESOURCE_CREATE_BLOB", "")

Originally reported at https://github.com/talex5/wayland-proxy-virtwl/issues/95
---
 rutabaga_gfx/src/cross_domain/mod.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rutabaga_gfx/src/cross_domain/mod.rs b/rutabaga_gfx/src/cross_domain/mod.rs
index a35bf6425..56c6b7282 100644
--- a/rutabaga_gfx/src/cross_domain/mod.rs
+++ b/rutabaga_gfx/src/cross_domain/mod.rs
@@ -344,7 +344,7 @@ impl CrossDomainWorker {
                                     )
                                 }
                                 Ok(DescriptorType::WritePipe) => {
-                                    *identifier_type = CROSS_DOMAIN_ID_TYPE_VIRTGPU_BLOB;
+                                    *identifier_type = CROSS_DOMAIN_ID_TYPE_WRITE_PIPE;
                                     add_item(
                                         &self.item_state,
                                         CrossDomainItem::WaylandWritePipe(WritePipe::new(
